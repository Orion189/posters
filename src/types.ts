import * as yup from 'yup';
import type { NextApiRequest } from 'next';
import { MovieType, PosterType, Forms } from 'src/enums';
import { validationConfig } from 'src/configs/validation.config';

export type Genre = {
    id: number;
    name: string;
};

export type Country = {
    country_code: string;
    country: string;
};

export type Poster = {
    id: number;
    group_id: number;
    country_code: string;
    file_location: string;
    width: number;
    height: number;
    size: number;
    type_id: PosterType;
};

export type Movie = {
    id: number;
    imdb: number;
    type: MovieType;
    year: number;
    original_title: string;
    url_title: string;
    num_posters: number;
    num_groups: number;
    language: string;
    country: string;
    release_date: string;
    director: string;
    genre: string;
    runtime: number;
    summary: string;
    poster?: Poster;
};

export type GraphQLServerContext = {
    req: NextApiRequest;
};

export type Settings = yup.InferType<(typeof validationConfig)[Forms.SETTINGS]>;

export type MovieInfo = Pick<
    Movie,
    'year' | 'original_title' | 'country' | 'director' | 'genre' | 'runtime' | 'summary'
>;
