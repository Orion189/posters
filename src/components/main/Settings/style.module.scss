.panelTitle {
    color: rgb(245, 245, 245);
    margin-bottom: 4rem;
}

.form {
    display: flex;
    flex-direction: column;
    padding: 2.5rem;
}
