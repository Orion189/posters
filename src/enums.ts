export const enum MovieType {
    BOTH = '0',
    MOVIE = '1',
    SERIES = '2'
}

export const enum PosterType {
    UNSET = 0,
    POSTER = 1,
    COVER = 2,
    TEXTLESS = 3,
    LOGO = 4,
    PROMO = 5,
    CUSTOM = 6,
    ADVANCE = 7,
    TEASER = 8,
    THEATRICAL = 9,
    RE_RELEASE = 10,
    COMBO = 11,
    VIDEO_RELEASE = 12,
    CONCEPT = 13,
    NEVER_PRINTED = 14
}

export const enum GenreType {
    DRAMA = 1,
    ROMANCE = 2,
    SCI_FI = 3,
    THRILLER = 4,
    HORROR = 5
}

export const enum PosterThumbSizeType {
    XL = 'xl',
    L = 'l',
    S = 's',
    T = 't',
    M = 'm'
}

export const enum PersonThumbSizeType {
    S = 's',
    T = 't',
    M = 'm'
}

export const enum SortType {
    ORIGINAL_TITLE = 'original_title',
    NUM_POSTERS = 'num_posters',
    YEAR = 'year',
    RUNTIME = 'runtime',
    COUNTRY = 'country'
}

export const enum DirectionType {
    ASC = 'asc',
    DESC = 'desc'
}

export const enum Forms {
    SETTINGS = 'Settings'
}
