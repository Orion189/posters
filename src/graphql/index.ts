import getConfig from 'next/config';
import { ApolloClient, InMemoryCache } from '@apollo/client';

const { publicRuntimeConfig } = getConfig();

const client = new ApolloClient({
    uri: `${publicRuntimeConfig.host}/api/graphql`,
    cache: new InMemoryCache()
});

export default client;
