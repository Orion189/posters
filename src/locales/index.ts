import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import detector from 'i18next-browser-languagedetector';

import * as en from './en.json';
import * as ru from './ru.json';

const resources = {
    en: { translation: en },
    ru: { translation: ru }
};

i18n.use(detector)
    .use(initReactI18next)
    .init({
        resources,
        fallbackLng: 'en',
        interpolation: {
            escapeValue: false // react already safes from xss
        }
    });

export default i18n;
