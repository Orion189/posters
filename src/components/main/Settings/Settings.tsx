import { FC } from 'react';
import { Formik, Form, FastField } from 'formik';
import Button from '@mui/material/Button';
import FormGroup from '@mui/material/FormGroup';
import FormControl from '@mui/material/FormControl';
import Typography from '@mui/material/Typography';
import { useTranslation } from 'react-i18next';
import SelectInput from '@components/shared/SelectInput';
import { Settings, Genre, Country } from 'src/types';
import { Forms, MovieType } from 'src/enums';
import { validationConfig, yearsList } from 'src/configs/validation.config';
import styles from './style.module.scss';
import commonStyles from '@styles/common.module.scss';

type SettingsProps = {
    genresData?: [Genre];
    countriesData?: [Country];
    onSaveForm: (values: Settings) => void;
    onResetForm: () => void;
};

const Settings: FC<SettingsProps> = ({ onSaveForm, genresData, countriesData, onResetForm }) => {
    const { t } = useTranslation();
    const movieTypeData = [
        { key: MovieType.BOTH, value: t(`common.label.MovieType.${MovieType.BOTH}`) },
        { key: MovieType.MOVIE, value: t(`common.label.MovieType.${MovieType.MOVIE}`) },
        { key: MovieType.SERIES, value: t(`common.label.MovieType.${MovieType.SERIES}`) }
    ];
    const countryTypeData =
        countriesData &&
        countriesData.map((country: Country) => ({
            key: country.country_code,
            value: country.country
        }));
    const genreTypeData =
        genresData &&
        genresData.map((genre: Genre) => ({
            key: genre.id.toString(),
            value: genre.name
        }));
    const yearData = yearsList.map((year: string) => ({
        key: year,
        value: year
    }));
    const initialValues = {
        movieType: MovieType.BOTH,
        genreType: '',
        yearFrom: '',
        yearTo: '',
        country: ''
    };

    return (
        <Formik
            onSubmit={onSaveForm}
            validateOnChange={false}
            enableReinitialize={true}
            initialValues={initialValues}
            validationSchema={validationConfig[Forms.SETTINGS]}
        >
            {({ resetForm }) => (
                <Form noValidate autoComplete="off" className={styles.form}>
                    <Typography variant="h5" gutterBottom className={styles.panelTitle}>
                        {t('components.main.Settings.title')}
                    </Typography>
                    <FormControl variant="standard">
                        <FastField
                            component={SelectInput}
                            name="movieType"
                            label={t('components.main.Settings.placeholder.movieType')}
                            data={movieTypeData}
                        />
                    </FormControl>
                    <FormControl variant="standard">
                        <FastField
                            component={SelectInput}
                            name="genreType"
                            label={t('components.main.Settings.placeholder.genreType')}
                            data={genreTypeData}
                        />
                    </FormControl>
                    <FormControl variant="standard">
                        <FastField
                            component={SelectInput}
                            name="yearFrom"
                            label={t('components.main.Settings.placeholder.yearFrom')}
                            data={yearData}
                        />
                    </FormControl>
                    <FormControl variant="standard">
                        <FastField
                            component={SelectInput}
                            name="yearTo"
                            label={t('components.main.Settings.placeholder.yearTo')}
                            data={yearData}
                        />
                    </FormControl>
                    <FormControl variant="standard">
                        <FastField
                            component={SelectInput}
                            name="country"
                            label={t('components.main.Settings.placeholder.country')}
                            data={countryTypeData}
                        />
                    </FormControl>
                    <FormGroup className={commonStyles.actionsCont}>
                        <FormControl variant="standard">
                            <Button type="submit" color="primary" variant="contained">
                                {t('components.main.Settings.label.save')}
                            </Button>
                        </FormControl>
                        <FormControl variant="standard">
                            <Button
                                color="primary"
                                variant="outlined"
                                onClick={() => {
                                    resetForm();
                                    onResetForm();
                                }}
                            >
                                {t('components.main.Settings.label.reset')}
                            </Button>
                        </FormControl>
                    </FormGroup>
                </Form>
            )}
        </Formik>
    );
};

export default Settings;
