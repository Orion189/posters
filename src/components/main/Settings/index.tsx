import { FC, useCallback } from 'react';
import getConfig from 'next/config';
import { useSettings } from '@components/hooks/useSettings';
import { Genre, Country, Settings } from 'src/types';
import SettingsView from './Settings';

type SettingsProps = {
    genresData?: [Genre];
    countriesData?: [Country];
};

const { publicRuntimeConfig } = getConfig();

const SettingsFC: FC<SettingsProps> = ({ genresData, countriesData }) => {
    const { setSettingsStored } = useSettings();
    const onSaveForm = useCallback(
        (values: Settings) => {
            window.localStorage.setItem(publicRuntimeConfig.lsItemName, JSON.stringify(values));

            setSettingsStored(true);
        },
        [setSettingsStored]
    );

    const onResetForm = useCallback(() => {
        window.localStorage.removeItem(publicRuntimeConfig.lsItemName);

        setSettingsStored(false);
    }, [setSettingsStored]);

    return (
        <SettingsView
            genresData={genresData}
            countriesData={countriesData}
            onSaveForm={onSaveForm}
            onResetForm={onResetForm}
        />
    );
};

export default SettingsFC;
