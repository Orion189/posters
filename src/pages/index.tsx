import { FC } from 'react';
import Head from 'next/head';
import Image from 'next/image';
import client from 'src/graphql';
// import getConfig from 'next/config';
import { GetStaticProps } from 'next';
import styles from './style.module.scss';
import type { Movie, Settings } from 'src/types';
import { useMovies } from '@components/hooks/useMovies';
import { GET_GENRES, GET_COUNTRIES } from 'src/graphql/queries';

type IndexPageProps = {
    settingsData: Settings;
};

// const { publicRuntimeConfig } = getConfig();
const getGenres = async () => {
    // const response = await fetch(`${publicRuntimeConfig.host}/api/genres`);
    // const genres = await data.json();
    try {
        const {
            data: { genres }
        } = await client.query({
            query: GET_GENRES
        });

        return genres;
    } catch {
        return [];
    }
};
const getCountries = async () => {
    // const response = await fetch(`${publicRuntimeConfig.host}/api/countries`);
    // const countries = await response.json();
    try {
        const {
            data: { countries }
        } = await client.query({
            query: GET_COUNTRIES
        });

        return countries;
    } catch {
        return [];
    }
};

export const getStaticProps: GetStaticProps = async () => {
    const genresData = await getGenres();
    const countriesData = await getCountries();

    if (!genresData || !countriesData) {
        return {
            notFound: true
        };
    }

    return {
        props: {
            genresData,
            countriesData
        }
    };
};

const IndexPage: FC<IndexPageProps> = () => {
    const { index, movies } = useMovies();

    return index !== undefined ? (
        <div className={styles.cont}>
            <Head>
                <title>{`${movies[index]?.original_title}, ${movies[index]?.year}`}</title>
                <meta name="description" content={movies[index]?.summary} />
                <link rel="icon" href="/favicon.ico" />
            </Head>
            {movies?.map(
                (mov: Movie) =>
                    mov.poster && (
                        <div
                            className={`${styles.wrapper} ${mov.id === movies[index]?.id ? '' : styles.hidden}`}
                            key={mov.id.toString()}
                        >
                            <Image
                                alt={mov.original_title}
                                src={mov.poster?.file_location}
                                width={mov.poster?.width}
                                height={mov.poster?.height}
                            />
                        </div>
                    )
            )}
        </div>
    ) : null;
};

export default IndexPage;
