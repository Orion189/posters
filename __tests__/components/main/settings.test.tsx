import { cleanup, render, fireEvent, act, waitFor } from '@testing-library/react';
import { Genre, Country } from 'src/types';
import Settings from '@components/main/Settings';

const genresData: [Genre] = [{
    id: 1,
    name: 'Drama'
}];
const countriesData: [Country] = [{
    country_code: 'US',
    country: 'United States'
}];
const values = {
    movieType: '0',
    genreType: '1',
    yearFrom: '1940',
    yearTo: '2000',
    country: 'US'
};
let setSettingsStored: jest.Mock;

jest.mock('react-i18next', () => ({
    useTranslation: () => ({
        t: jest.fn((str) => str),
        i18n: {
            changeLanguage: () => new Promise(() => {})
        }
    })
}));

jest.mock('@components/hooks/useSettings', () => ({
    useSettings: () => ({ setSettingsStored })
}));

describe('Settings', () => {
    beforeEach(() => {
        window.localStorage.clear();
        setSettingsStored = jest.fn();
        jest.spyOn(Object.getPrototypeOf(window.localStorage), 'setItem');
    });

    afterEach(() => {
        cleanup();
        jest.restoreAllMocks();
    });

    it('Saving preferences', async () => {
        act(() => {
            render(<Settings genresData={genresData} countriesData={countriesData} />);
        })

        const movieType = document.querySelector('[name="movieType"]');
        const genreType = document.querySelector('[name="genreType"]');
        const yearFrom = document.querySelector('[name="yearFrom"]');
        const yearTo = document.querySelector('[name="yearTo"]');
        const country = document.querySelector('[name="country"]');
        const submit = document.querySelector('[type="submit"]');

        act(() => {
            fireEvent.change(movieType!, { target: { value: values.movieType } });
            fireEvent.change(genreType!, { target: { value: values.genreType } });
            fireEvent.change(yearFrom!, { target: { value: values.yearFrom } });
            fireEvent.change(yearTo!, { target: { value: values.yearTo } });
            fireEvent.change(country!, { target: { value: values.country } });

            fireEvent.click(submit!);
        });
        
        await waitFor(() => {
            expect(setSettingsStored).toHaveBeenCalledWith(true);
            expect(window.localStorage.setItem).toHaveBeenCalledWith('settings', JSON.stringify(values));
        });     
    });
/*
    it('renders <Settings /> props', () => {
        render(<Settings />);

        const original_title = screen.getByText(movieInfo.original_title);
        const year = screen.getByText(movieInfo.year);
        const country = screen.getByText(movieInfo.country);
        const director = screen.getByText(movieInfo.director);
        const genre = screen.getByText(movieInfo.genre);
        const runtime = screen.getByText(movieInfo.runtime);
        const summary = screen.getByText(movieInfo.summary);

        expect(original_title).toBeInTheDocument();
        expect(year).toBeInTheDocument();
        expect(country).toBeInTheDocument();
        expect(director).toBeInTheDocument();
        expect(genre).toBeInTheDocument();
        expect(runtime).toBeInTheDocument();
        expect(summary).toBeInTheDocument();
    });
*/
    it('Renders <Settings /> unchanged', () => {
        const { container } = render(<Settings genresData={genresData} countriesData={countriesData} />);

        expect(container).toMatchSnapshot()
    });
});
