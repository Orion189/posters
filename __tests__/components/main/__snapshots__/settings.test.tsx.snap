// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Renders <Settings /> unchanged 1`] = `
<div>
  <form
    action="#"
    autocomplete="off"
    class="form"
    novalidate=""
  >
    <h5
      class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom panelTitle css-h93ljk-MuiTypography-root"
    >
      components.main.Settings.title
    </h5>
    <div
      class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        id="movieType"
      >
        components.main.Settings.placeholder.movieType
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl  css-a3l6o-MuiInputBase-root-MuiInput-root-MuiSelect-root"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="movieType mui-component-select-movieType"
          class="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
          id="mui-component-select-movieType"
          role="button"
          tabindex="0"
        >
          common.label.MovieType.0
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          name="movieType"
          tabindex="-1"
          value="0"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
    <div
      class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        id="genreType"
      >
        components.main.Settings.placeholder.genreType
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl menuItemDefaultValue css-a3l6o-MuiInputBase-root-MuiInput-root-MuiSelect-root"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="genreType mui-component-select-genreType"
          class="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
          id="mui-component-select-genreType"
          role="button"
          tabindex="0"
        >
          <span
            class="notranslate"
          >
            ​
          </span>
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          name="genreType"
          tabindex="-1"
          value=""
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
    <div
      class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        id="yearFrom"
      >
        components.main.Settings.placeholder.yearFrom
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl menuItemDefaultValue css-a3l6o-MuiInputBase-root-MuiInput-root-MuiSelect-root"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="yearFrom mui-component-select-yearFrom"
          class="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
          id="mui-component-select-yearFrom"
          role="button"
          tabindex="0"
        >
          <span
            class="notranslate"
          >
            ​
          </span>
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          name="yearFrom"
          tabindex="-1"
          value=""
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
    <div
      class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        id="yearTo"
      >
        components.main.Settings.placeholder.yearTo
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl menuItemDefaultValue css-a3l6o-MuiInputBase-root-MuiInput-root-MuiSelect-root"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="yearTo mui-component-select-yearTo"
          class="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
          id="mui-component-select-yearTo"
          role="button"
          tabindex="0"
        >
          <span
            class="notranslate"
          >
            ​
          </span>
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          name="yearTo"
          tabindex="-1"
          value=""
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
    <div
      class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        id="country"
      >
        components.main.Settings.placeholder.country
      </label>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl menuItemDefaultValue css-a3l6o-MuiInputBase-root-MuiInput-root-MuiSelect-root"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="country mui-component-select-country"
          class="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
          id="mui-component-select-country"
          role="button"
          tabindex="0"
        >
          <span
            class="notranslate"
          >
            ​
          </span>
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          name="country"
          tabindex="-1"
          value=""
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
    <div
      class="MuiFormGroup-root actionsCont css-dmmspl-MuiFormGroup-root"
    >
      <div
        class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          components.main.Settings.label.save
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          components.main.Settings.label.reset
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </form>
</div>
`;
